### PR TITLE
Add an example for automm distillation.

### PR DESCRIPTION
**Pull request by FANGAreNotGnu**
*Description of changes:*
Add an example for distillation using GLUE datasets.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: examples/automm/distillation/README.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+Examples showing how to use AutoMMPredictor distillation.
```
Comment (sxjscience) [inline examples/automm/distillation/README.md:1]: Consider to change to 

```
Examples showing how to use `AutoMMPredictor` for model distillation.
```

File: examples/automm/distillation/README.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+### 1.1 Example
+[`automm_distillation.py`](./automm_distillation.py) : This example provide a use case 
+for using distillation in AutoMMPredictor, as well as the evaluation of the distillation result.
+
+To run the example:
+
+```python automm_distillation.py --glue_task stsb --<other_flag> <value>```
+   - `glue_task` determines to run the experiments on which GLUE task, refers to [Dataset Section](###1.2-Datasets).
+   - `teacher_model` is the name of teacher model
+   - `student_model` is the name of the student model. It is recommended to have the same backbone as teacher model (e.g. both Electra or both BERT).
```
Comment (sxjscience) [inline examples/automm/distillation/README.md:14]: nit: "Electra" should be "ELECTRA".

File: examples/automm/distillation/automm_distillation.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--glue_task", default="qnli", type=str)
+    parser.add_argument("--teacher_model", default="google/bert_uncased_L-12_H-768_A-12", type=str)
+    parser.add_argument("--student_model", default="google/bert_uncased_L-4_H-768_A-12", type=str)
+    parser.add_argument("--seed", default=123, type=int)
+    parser.add_argument("--max_epochs", default=1000, type=int)
+    parser.add_argument("--time_limit", default=7200, type=int)
```
Comment (sxjscience) [inline examples/automm/distillation/automm_distillation.py:138]: We can also use `None` as the default time-limit so these models will be trained until convergence.

File: examples/automm/distillation/README.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            teacher_output=teacher_output,
+        )
+        loss += soft_label_loss * self.soft_label_weight
+```
+
+### 1.4 Performance
+```
+glue_task = qnli
+Teacher Model = google/bert_uncased_L-12_H-768_A-12
+Pretrained Model = google/bert_uncased_L-6_H-768_A-12
```
Comment (sxjscience) [inline examples/automm/distillation/README.md:58]: Consider to change to "Student Model".

File: examples/automm/distillation/README.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+seed = 123
+max_epoch = 8
+time_limit = 3600
+metric = "accuracy"
```
Comment (sxjscience) [inline examples/automm/distillation/README.md:62]: We can also attach an example command like

```bash
python3 automm_distillation.py --teacher_model google/bert_uncased_L-12_H-768_A-12 --student_model google/bert_uncased_L-6_H-768_A-12 --seed 123 
```

**Comment by sxjscience** [comment]
Added a few comment. Overall looks good! We can also attach an example about model distillation!

**Review by zhiqiangdon — APPROVED** [review]

File: examples/automm/distillation/README.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+   - `num_gpu` determines num of gpu used to train the models. Default is -1 which means using all.
+   - `temperature` determines the temperature for soft cross entropy.
+   - `hard_label_weight` determines the weight of hard labels (ground truth logits).
+   - `soft_label_weight` determines the weight of soft labels (teacher model's output).
+
+### 1.2 Dataset
+We borrow 7 NLP tasks in GLUE [1], and use identically the same abbreviation as in [1] to name each tasks,
+i.e. "mnli(m/mm)", "qqp", "qnli", "sst2", "stsb", "mrpc", and "rte".
+The dataset are loaded using huggingface's API: https://huggingface.co/datasets/glue.
+All Data will be automatically downloaded from HuggingFace (thus online connection is necessary) if it does not exisit with the given dataset path.
```
Comment (bryanyzhu) [inline examples/automm/distillation/README.md:27]: exisit -> exist

File: examples/automm/distillation/README.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
+Examples showing how to use `AutoMMPredictor` for model distillation.
+
+## 1. Distillation on GLUE tasks
+
+### 1.1 Example
+[`automm_distillation.py`](./automm_distillation.py) : This example provide a use case
```
Comment (bryanyzhu) [inline examples/automm/distillation/README.md:6]: provide -> provides

**Review by bryanyzhu — APPROVED** [review]
LGTM

File: examples/automm/distillation/README.md (hunk 7 above)
Comment (sxjscience) [reply examples/automm/distillation/README.md:6]: Have merged for now but @FANGAreNotGnu feel free to submit subsequent PRs to fix the typos! Thanks!

Comment (FANGAreNotGnu) [reply examples/automm/distillation/README.md:6]: I'll fix the typo in next PR, and also provide more experiment results. Thanks!

**Pull request merged**